Make Teleport version requirements more consistent

ptgott opened the pull request:
Closes #40886

When a how-to guide mentions the need for a specific Teleport version, ensure that the Prerequisites section of the guide includes the version requirement. This keeps all prerequisites for the guide in one place and avoids surprising the user with a version requirement at an unexpected location within the guide.

Also remove mentions of Teleport versions before v13, which are unsupported in the docs. Don't touch reference guides, which we do not expect to include a Prerequisites section.

ptgott committed:
Make Teleport version requirements more consistent

Closes #40886

When a how-to guide mentions the need for a specific Teleport version,
ensure that the Prerequisites section of the guide includedes the
version requirement. This keeps all prerequisites for the guide in one
place and avoids surprising the user with a version requirement at an
unexpected location within the guide.

Also remove mentions of Teleport versions before v13, which are
unsupported in the docs. Don't touch reference guides, which we do not
expect to include a Prerequisites section.

zmb3 commented on docs/pages/access-controls/device-trust/guide.mdx:
````diff
@@ -45,54 +45,53 @@ protected with Teleport.
   root@(=clusterDefaults.nodeIP=):~#
   ```
 
+<Details type="warning" title="Teleport v13.3.5 and Below">
````
Do we need to mention v13 anymore?

ptgott replied:
Our convention has been to remove version warnings for versions below the latest EOL version: https://github.com/gravitational/teleport/blob/master/.github/ISSUE_TEMPLATE/test-plan-docs.md#is-the-docs-site-up-to-date-with-the-new-release

zmb3 commented on docs/pages/access-controls/guides/headless.mdx:
```diff
@@ -25,17 +25,17 @@ For example:
   See the [Second Factor: WebAuthn](./webauthn.mdx) guide.
 - WebAuthn hardware device, such as YubiKey.
-- Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-and-tctl-clients-only) `tsh` binary v12.2+ installed.
+- Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-client-only) `tsh` binary installed.
```
This looks like a regression (broken link).

mmcallister replied:
It looks like `## Windows (tsh and tctl clients only)` is the correct header

ptgott committed:
Address zmb3 feedback

- Fix link
- Remove paragraph that pertains to v12

nklaassen commented on docs/pages/includes/tls-multiplexing-warnings.mdx:
````diff
@@ -1,52 +0,0 @@
-submit a [GitHub
-issue](https://github.com/gravitational/teleport/issues/new/choose).
-
-You must ensure that your reverse proxy is communicating with Teleport using HTTPS.
-When running Teleport in Kubernetes and using nginx as an ingress, this requires adding
-an annotation to the chart values:
-
-```yaml
-annotations:
-  ingress:
-    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
-```
-
-Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
-(`cloudflared`) should work with Teleport version 15.1 or higher. See the
-[TLS Routing FAQ](../reference/architecture/tls-routing.mdx) for more details.
````
this section seems relevant for 14.0+, should we keep it and delete the rest?

ptgott committed:
Respond to nklaassen feedback

Keep `tls-multiplexing-warnings.mdx`, preserving only content that is
relevant for v14+.

ptgott commented:
@nklaassen Following up to see if I have resolved your feedback. Thanks!

mmcallister commented on docs/pages/admin-guides/access-controls/device-trust/guide.mdx:
````diff
@@ -45,54 +45,53 @@ protected with Teleport.
+  $ tctl edit role/access
+  ```
+
+  Edit the role with device trust mode:
````
```suggestion
  Edit the role with Device Trust mode:
```

ptgott committed:
Update docs/pages/admin-guides/access-controls/device-trust/guide.mdx

Co-authored-by: M.C.M. <marie.mcallister@goteleport.com>
